### PR TITLE
SNOW-10: Enable categories on pages and show the navigational sidebar on the About Us page

### PR DESCRIPTION
**Pull request by grrrero**
JIRA: [SNOW-10](https://issues.fluidproject.org/browse/SNOW-10)

(This was renamed to SNOW-10 from SNOW-7 as the latter is the branch for CSS refactoring.)

From [Stack Overflow](https://stackoverflow.com/a/35087335)

In order for Wordpress to enable categories (and tags) support for pages, we need to register the taxonomy for those objects. Taxonomies refer to a way of grouping objects, and in our case, we want to group the object type of pages.

After that they need to be included in the queries (access $query object) so that Wordpress knows to interact with this new addition to its taxonomy.

Lastly, add_action is the hook that will launch the functions. The add_action hook with tag 'init' initialises the function 'tags_categories_support_all' at the time that Wordpress loads. 'pre_get_posts' gives the function access to the $query object (which is passed by reference, meaning they don't have to be declared or returned).

I renamed the functions and put the add_action hooks after their corresponding functions for easier reading. 

```
// Add tag and category support to pages
function allow_pages_tags_categories() {
 register_taxonomy_for_object_type('post_tag', 'page');
 register_taxonomy_for_object_type('category', 'page'); 
}
add_action('init', 'allow_pages_tags_categories');
// Ensure all tags and categories are included in queries
function add_queries_tags_categories($wp_query) {
 if ($wp_query->get('tag')) $wp_query->set('post_type', 'any');
 if ($wp_query->get('category_name')) $wp_query->set('post_type', 'any');
}
add_action('pre_get_posts', 'add_queries_tags_categories');
```

### CODE REVIEW

**Comment by cindyli** [comment]
BTW, the link to the JIRA seems wrong. It should be https://issues.fluidproject.org/browse/SNOW-10

File: functions.php [hunk 1]
```diff
@@ -165,43 +127,120 @@ public function widget( $args ) {
...
+
+function create_sidebar($post_type) {
+    $category_current = get_the_category();
+    $category_link =  get_term_link ($category_current[0]->slug, "category");
+    $current_displayed_id = get_the_ID (); // the ID of the post currently displayed in the main content panel.
+
+    if ( ! empty( $category_current ) ) {
+        $args = array(
+            // get posts belonging to the current category.
+            'post_type' => '$post_type',
```
Comment (cindyli) [inline functions.php]: Is it necessary to have quotes(') around a variable?

File: functions.php [hunk 2]
```diff
@@ -165,43 +127,120 @@ public function widget( $args ) {
...
+                </ul>
+                <?php
+            endif;
+        endif;
+    }
+}
+
+// Add tag and category support to pages
+function allow_pages_tags_categories() {
+  register_taxonomy_for_object_type('post_tag', 'page');
```
Comment (cindyli) [inline functions.php]: Reduce the duplication of the code at line 233, 234 with a loop. The same comment applies to line 240, 241.

**Comment by grrrero** [comment]
Hi @cindyli I've made the changes based on your review.

**Comment by cindyli** [comment]
Please fix indentations to use 4 spaces. Thanks.

**Comment by grrrero** [comment]
Will do @cindyli

**Comment by cindyli** [comment]
This pull request looks good and can be merged into the master. Thanks, @grrrero.

**Comment by grrrero** [comment]
Thank you so much @cindyli  (ﾉ◕ヮ◕)ﾉ*:･ﾟ✧

**Pull request merged**